crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -51,6 +56,209 @@ use sui_types::sui_serde::{Base64, Encoding};
 mod rpc_types_tests;
 
 pub type GatewayTxSeqNumber = u64;
+pub type SuiMoveTypeParameterIndex = u16;
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct SuiMoveAbilitySet(u8);
```

Comment:
This should not be `u8`. They should be serialized to `Drop`, `Store`, etc

Reply:
Drop and Store are Ability and AbilitySet is different, the original definition is here
https://github.com/move-language/move/blob/8958b40e97a4f94a588e5365fdb90ec18f00d888/language/move-binary-format/src/file_format.rs#L599

Reply:
`AbilitySet` is a set of `Ability` represented by bit mask: you can serialize it a list of `Ability` by doing something like https://github.com/move-language/move/blob/8958b40e97a4f94a588e5365fdb90ec18f00d888/language/move-binary-format/src/file_format.rs#L776-L784

Reply:
I see what you meant, indeed I can convert it back to a Set, but if we want to do ability check, we can add similar impl like this later while still keeping this struct itself?
https://github.com/move-language/move/blob/8958b40e97a4f94a588e5365fdb90ec18f00d888/language/move-binary-format/src/file_format.rs#L601

Reply:
I would avoid duplicating ability definition on TS side(so that we don't need to worry about out of sync issues), therefore it's simpler to just return the string representation.